localstripe/resources.py

```diff
@@ -3144,6 +3144,30 @@ def _calculate_amount_in_tier(self, quantity, index):
         t = self.plan.tiers[index]
         return int(t['unit_amount']) * quantity + int(t['flat_amount'])
 
+    @classmethod
+    def _api_list_all(cls, url, subscription=None, limit=None,
+                      starting_after=None, ending_before=None, **kwargs):
+        if kwargs:
+            raise UserError(400, 'Unexpected ' + ', '.join(kwargs.keys()))
+
+        # to return 404 if not existant
+        Subscription._api_retrieve(subscription)
+
+        try:
+            assert _type(subscription) is str
+            assert subscription.startswith('sub_')
+        except AssertionError:
+            raise UserError(400, 'Bad request')
+
+        li = super(SubscriptionItem, cls)._api_list_all(
+            url, limit=limit, starting_after=starting_after
+        )
+
+        li._list = [obj for obj in li._list
+                    if obj.subscription == subscription]
```

Comment:
A quick diff proposition here:
- don't add a new line before `)` so we save one line
- don't add a new line after `)` so we save one line, and it's more similar to the other API

```diff
--- a/localstripe/resources.py
+++ b/localstripe/resources.py
@@ -3160,9 +3160,7 @@ class SubscriptionItem(StripeObject):
             raise UserError(400, 'Bad request')
 
         li = super(SubscriptionItem, cls)._api_list_all(
-            url, limit=limit, starting_after=starting_after
-        )
-
+            url, limit=limit, starting_after=starting_after)
         li._list = [obj for obj in li._list
                     if obj.subscription == subscription]